pkg/server/proto/server.proto

```diff
@@ -619,6 +623,12 @@ message Variable {
     HclPos start = 2;
     HclPos end = 3;
   }
+
+  message Ref {
+    string value = 2;
```

Comment:
It feels like this message should be a oneof...

I feel like this is slightly more analogous to `message Variable` too.

```proto
message Ref {
  Source source = 1;

  enum Source {
    // TODO: comments
    UNKNOWN = 0;
    FILE = 1;
    // etc...
  }

  oneof value {
    // sensitive values just have a salted hash of the value. If you have the
    // secret value, you can SHA-256 hash it with the server cookie as a
    // salt to verify this value matched.
    string sensitive = 1;

    // The raw value of this variable. 
    string str = 1;
    bool bool = 1;
    int64 num = 1;
    string hcl = 1;
  }
}
```

Reply:
Also, it seems like `sensitive` is duplicated here from `Variable` but I think they serve two diff purposes:

* The `sensitive` on `Variable` notes that that variable is _currently_ sensitive. This can change at any point from false to true or true to false.
* The `sensitive` on `Ref` denotes the variable was sensitive _at the time_ the value was rendered, and also uses the protobuf oneof and type system to make it clear the value is the hash (versus the `string str` field). 
  - If at a future point, a variable is updating to no longer be sensitive, past rendered values are still sensitive. So you can't rely on the `Variable.sensitive` field to let you know about this. You have to encode it in the `Ref`.